app/controllers/claims_controller.rb

```diff
@@ -0,0 +1,44 @@
+class ClaimsController < ApplicationController
+  def new
+    contribution = Listing.find(params[:contribution_id])
+    if contribution.person&.email.blank?
+      flash.now[:alert] = "We are sorry, this contributor hasn't provided
+                         an email address yet and can't communicate this way".squish
+    end
+    render locals: { contribution: contribution }
+  end
+
+  def create
+    # TODO: Need to handle race conditions to prevent creating multiple matches for same contribution.
+
+    contribution = Listing.find(params[:contribution_id])
+    ActiveRecord::Base.transaction do
+      if current_user.person.blank?
+        Person.create_from_peer_to_peer_params!(current_user, name: peer_to_peer_match_params[:peer_alias],
+                                                preferred_contact_method_id: peer_to_peer_match_params[:preferred_contact_method_id],
+                                                contact_info: peer_to_peer_match_params[:preferred_contact_details])
+      end
```

Comment:
I think we might also need to handle when the current_user _changes_ their contact preference and info.
IIUC, such a change would currently be ignored.

Reply:
Yes, you're right. That change is completely ignored as of now. We're only considering email. My take would be to create a notification interface and depending on the contact preference method, we can either notify via an email or SMS.

Do you want me to make any specific changes here or to the form to restrict only to email?

Reply:
We can discuss more at our 9am if you like, but i'm leaning toward only allowing email on the form for this iteration.
In the controller, if we have to fabricate a person instance, we can force email to be the preferred contact method.

In addition, it would be good to save the email address if the person already existed but in this case, maybe leave their preference alone, just updating their email field?

Reply:
@exbinary I have made changes to restrict the preferred contact method to just email for now and also saving email to the person record if absent in the system. Let me know how that looks. 

I'm gonna branch off of this branch and start working on implementing `active_interaction`. I thought that shouldn't stop us from merging these changes.

Reply:
Looks great 🚀 . Agree on making the interactor refactoring a separate branch 👍🏾.